dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -15,21 +15,28 @@ sealed class TypedColumn[T, U](
 )(
   implicit
   val uencoder: TypedEncoder[U]
-) extends UntypedExpression[T] {
+) extends UntypedExpression[T] { self =>
 
   def this(column: Column)(implicit uencoder: TypedEncoder[U]) {
     this(FramelessInternals.expr(column))
   }
 
   def untyped: Column = new Column(expr)
 
-  def ===(other: U): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other)
-  }
+  def ===(other: U): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other)
+  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other.untyped)
 
-  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other.untyped)
-  }
+  def +(u: TypedColumn[T, U])(implicit n: Numeric[U]):  TypedColumn[T, U]= new TypedColumn[T, U](self.untyped.plus(u.untyped))
```

Comment:
I wonder if we should use our own type class here instead of `Numeric`. IIUC it's only used to constraint the types on which `+` is available, but users might be surprised if this does not work for something such a `BigDecimal` even it from a Scala POV this type is numeric. People could also get confused if they try to call `+` on a String column, see the implicit not found error, and wrongly try to fix it with a custom `Numeric[String]` instance.

What I'm proposing would be define a `sealed trait NumericColumnForSpark[T]` with instances for each of the types we can verify in test that they are actually handled correctly by Spark.